single-line-diagram/single-line-diagram-core/src/test/java/com/powsybl/sld/iidm/TestUnknownComponent.java

```diff
@@ -0,0 +1,227 @@
+import org.w3c.dom.Element;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author Thomas Adam <tadam at silicom.fr>
+ */
+public class TestUnknownComponent extends AbstractTestCaseIidm {
+
+    private VoltageLevel vl1;
+
+    @Before
+    public void setUp() {
+        layoutParameters.setCellWidth(80);
+
+        network = Network.create("testCase11", "test");
+        graphBuilder = new NetworkGraphBuilder(network);
+
+        substation = createSubstation(network, "subst", "subst", Country.FR);
+
+        buildNetwork();
+    }
+
+    @Override
+    protected ResourcesComponentLibrary getResourcesComponentLibrary() {
+        return Mockito.spy(new ConvergenceComponentLibrary());
+    }
+
+    @Test
+    public void test() {
+        layoutParameters.setAddNodesInfos(true);
+
+        // build voltage level 1 graph
+        VoltageLevelGraph g1 = graphBuilder.buildVoltageLevelGraph(vl1.getId());
+
+        voltageLevelGraphLayout(g1);
+
+        // Make all 3wt unknown SVG component
+        Map<String, List<Element>> unknownSvgElements = componentLibrary.getSvgElements(ComponentTypeName.UNKNOWN_COMPONENT);
+        Optional<String> unknownComponentStyleClass = componentLibrary.getComponentStyleClass(ComponentTypeName.UNKNOWN_COMPONENT);
+        ComponentSize unknownSize = componentLibrary.getSize(ComponentTypeName.UNKNOWN_COMPONENT);
+        Map<Orientation, Component.Transformation> unknownTransformations = componentLibrary.getTransformations(ComponentTypeName.UNKNOWN_COMPONENT);
+        List<AnchorPoint> unknownAnchorPoints = componentLibrary.getAnchorPoints(ComponentTypeName.UNKNOWN_COMPONENT);
+        Optional<String> unknownSubComponentStyleClass = Optional.empty();
+
+        Mockito.when(componentLibrary.getSvgElements(ComponentTypeName.THREE_WINDINGS_TRANSFORMER)).thenReturn(unknownSvgElements);
+        Mockito.when(componentLibrary.getComponentStyleClass(ComponentTypeName.THREE_WINDINGS_TRANSFORMER)).thenReturn(unknownComponentStyleClass);
+        Mockito.when(componentLibrary.getSize(ComponentTypeName.THREE_WINDINGS_TRANSFORMER)).thenReturn(unknownSize);
+        Mockito.when(componentLibrary.getTransformations(ComponentTypeName.THREE_WINDINGS_TRANSFORMER)).thenReturn(unknownTransformations);
+        Mockito.when(componentLibrary.getAnchorPoints(ComponentTypeName.THREE_WINDINGS_TRANSFORMER)).thenReturn(unknownAnchorPoints);
+        Mockito.when(componentLibrary.getSubComponentStyleClass(ComponentTypeName.THREE_WINDINGS_TRANSFORMER, "WINDING1")).thenReturn(unknownSubComponentStyleClass);
+        Mockito.when(componentLibrary.getSubComponentStyleClass(ComponentTypeName.THREE_WINDINGS_TRANSFORMER, "WINDING2")).thenReturn(unknownSubComponentStyleClass);
+        Mockito.when(componentLibrary.getSubComponentStyleClass(ComponentTypeName.THREE_WINDINGS_TRANSFORMER, "WINDING3")).thenReturn(unknownSubComponentStyleClass);
```

Comment:
You should use mockito to simulate one component missing, so you should only need:
```java
Mockito.when(componentLibrary.getSvgElements(ComponentTypeName.THREE_WINDINGS_TRANSFORMER)).thenReturn(null);
```

Reply:
In ResourcesComponentLibrary :
We also need to simulate returning `null` from calls of : `components.get(ComponentTypeName.THREE_WINDINGS_TRANSFORMER)`
Or returning 'unknown comonent' as following : components.get(ComponentTypeName.UNKNOWN_COMPONENT) ?